examples/example-jacobi.cpp

```diff
@@ -0,0 +1,296 @@
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For additional details, please also read RAJA/LICENSE.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the disclaimer below.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+#include "memoryManager.hpp"
+
+const int CUDA_BLOCK_SIZE  = 16;
+const int CUDA_REDUCE_SIZE = 256;
+
+
+/*
+  Example 3: Jacobi Method
+
+  ----[Details]--------------------
+  This codes uses a five point finite difference stencil
+  to discretize 
+  
+  -U_xx - U_yy = 1.0.
+  The solution is assumed to satisfy
+  U = 0 on the boundary of the domain.
+
+  Values inside the domain are computed
+  using the Jocabi method to solve the associated
+  linear system. The domain is assumed to be a lattice 
+  with unit grid spacing.
+
+
+  ----[RAJA Concepts]---------------
+  1. RAJA ForallN
+  2. RAJA Reduction
+  3. RAJA::omp_collapse_nowait_exec
+  3. Reducing length of RAJA statements
+*/
+
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  printf("Example 3: Jacobi Method \n");
+
+  // Jacobi method is invoked until the l_2 norm of the difference of the
+  // iterations are under a tolerance
+  double tol = 1e-5;
+
+  int N       = 100;                // Number of unknown gridpoints per cartesian dimension
```

Comment:
This is a nitpick, but can you remove all the extra alignment spacing here and below - I think we want the RAJA code to have a uniform feel across the project.